Search

```diff
@@ -0,0 +1,45 @@
+// Time Complexity : I am weak in time/space complexities, I need some more learning on this. 
```

Comment:
Since you are reducing the search space by half, the time complexity is O(log n).
No additional data structures used, space complexity is O(1).

You will pick it up soon enough. Although,I would recommend you to submit your guess of the complexities in the PRs as well.